components/server/src/workspace/workspace-starter.ts

```diff
@@ -1302,6 +1307,11 @@ export class WorkspaceStarter {
                 }
             }
 
+            // Register a successful image build only if the image actually needed to be built; ie the build was not a no-op.
+            if (result.actuallyNeedsBuild) {
```

Comment:
One way to avoid this case would be to export both the `started` and `completed` metrics with a label of `needed_build`. This would still keep it semantic, and would give you the ability to filter these out in the SLO.